extensions/ql-vscode/src/quickEvalCodeLensProvider.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  CodeLensProvider,
+  TextDocument,
+  CodeLens, 
+  Command,
+  Range
+} from 'vscode';
+  
+class QuickEvalCodeLensProvider implements CodeLensProvider {
+  // Each provider requires a provideCodeLenses function which will give the various documents
+  // the code lenses
+  async provideCodeLenses(document: TextDocument): Promise<CodeLens[]> {
+
+    const codeLenses: CodeLens[] = [];
+
+    for (let index = 0; index < document.lineCount; index++) {
+      const textLine = document.lineAt(index);
+      // Match a predicate signature, including predicate name, parameter list, and opening brace.
+      const regex = new RegExp(/(\w+)\s*\(\s*.*(?:,\s*)*\)\s*\{/);
+      const matches = textLine.text.match(regex);
+      const startIndex = textLine.text.search(regex);
+
+      if (startIndex !== -1) {
```

Comment:
Avoid calling both `match` and `search`, because you're doubling the computation cost at each line by regex matching twice. Now that we're using information from the result of `match`, stick with that.
`matches.index` will give you the start index you need, and `if (matches)` will give you the condition to check whether the match succeeded (because it returns `null` if there is no match.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Reply:
Thanks for pointing this out. I got it fixed now. :)